SDK.md

```diff
@@ -0,0 +1,173 @@
+
+## Contribution acceptance
+
+Being an open source community CloudEvents team is open for a new members as well open to their contribution.
+In order to ensure that an SDK is going to be supported and maintained CloudEvents community would like to ensured that:
+
+ - a person (developer, committer) is going to become a maintainer
+ - a person commits to support ongoing changes to the [CloudEvent spec](spec.md)
+
+## Officially maintained software development kits (SDK)
+
+Software Development Kit (SDK) is a set of helpers designed and implemented to enhance and speed up a CloudEvents integration.
+As part of community efforts CloudEvents team committed to support and maintain the following SDKs:
+
+ - [Go SDK](https://github.com/cloudevents/sdk-go)
+ - [Java SDK](https://github.com/cloudevents/sdk-java)
+ - [Python SDK](https://github.com/cloudevents/sdk-python)
+ - [CSharp](https://github.com/cloudevents/sdk-csharp)
+ - [JavaScript SDK](https://github.com/cloudevents/sdk-javascript)
+
+## SDK contribution
+
+That's said, being an open source community CloudEvents is all about building open for contribution.
+In order to keep the development/user experience the same (as much as possible) CloudEvents team agreed to establish minimum requirements for a new SDK.
+
+### Technical requirements
+
+Supports CloudEvents spec milestones and ongoing development version.
+HTTP transport support (both structured and binary).
+Widely used programming language version.
+
+
+### Preferable API signature guidelines
+
+In order to remain developer/user UX the same among existing officially supported SDKs CloudEvents team asks maintainers to align with the following API signatures.
+Please consider the following code as pseudo-code.
+
+#### Event object constructor API
+
+Event build considered to be an event constructor:
+```
+    v01.Event()
+```
+
+#### Event object setters API 
+
+This particular code sample represents bare minimum number of setters:
+```
+    v01.Event().
+    WithContentType("application/json").
+    WithData('{"name":"john"}').
+    WithEventID("my-id").
+    WithSource("from-galaxy-far-far-away").
+    WithEventTime("tomorrow").
+    WithEventType("cloudevent.greet.you")
+```
+
+Content type setter represents an event MIME content type setter:
+```
+    WithContentType(content_type string)
+```
+
+Data setter represents an event data setter:
+```
+    WithData(event_data serializable)
+```
+
+ID setter represents an event ID setter:
+```
+    WithEventID(id string)
+```
+
+Source setter represents an event source setter:
+```
+    WithSource(source URL)
+```
+
+Time setter represents event emit time setter:
+```
+    WithEventTime(time RFC3339)
+```
+
+Type setter represents an event type setter:
+```
+    WithEventType(type string)
+```
+
+Extensions setter represents an event type setter:
+```
+    WithExtensions(exts map[string]string)
```

Comment:
May want to allow for setting an individual extension instead of all at once.

Reply:
unless you consider the generic Set() below to do that - in which case ignore my previous comment.

Reply:
yeah,`Set` would take care of a single extension, will a note about that.